src/vs/workbench/browser/parts/views/viewPaneContainer.ts

```diff
@@ -718,8 +719,9 @@ export class ViewPaneContainer extends Component implements IViewPaneContainer {
 	private computeInitialSizes(): Map<string, number> {
 		const sizes: Map<string, number> = new Map<string, number>();
 		if (this.dimension) {
-			const totalWeight = this.viewContainerModel.visibleViewDescriptors.reduce((totalWeight, { weight }) => totalWeight + (weight || 20), 0);
-			for (const viewDescriptor of this.viewContainerModel.visibleViewDescriptors) {
+			const shownViewDescriptors = this.viewContainerModel.visibleViewDescriptors.filter(viewDescriptor => viewDescriptor.when?.evaluate(this.contextKeyService.getContext(null)));
+			const totalWeight = shownViewDescriptors.reduce((totalWeight, { weight }) => totalWeight + (weight || 20), 0);
```

Comment:
@sbatten, there's a functional change here that I want to run by you before this is merged. Instead of computing the initial size across all views in a container, this now evaluates `when`s and only uses those that are currently visible.

Reply:
The previous evaluation was using `visibleViewDescriptors`, does your change suggest that there are views that are not visible in that list?

Reply:
cc @sandy081 as a co-author of the model

Reply:
Yes, I did find taht there are views taht are not visible in that list. That list seems to define visible as "checked in the view container ... list" and doesn't pay attention to `when` contexts, which can cause a view to be functionally non-existent as opposed to unchecked in the ... list.

Reply:
Hmm... that seems strange to me. I think there might be more to it. Can you share the case where this happens? For my investigation, I set a when context on the timeline view that makes it only present when a typescript file is open. Below is the list of Active and Visible view descriptors in the model. The red circle is with shows the timeline present in both list when its when clause is true. The second shows it removed from both lists when the when clause is false.

<img width="198" alt="image" src="https://user-images.githubusercontent.com/6561887/180062272-9ecb3235-3be5-405f-b9dd-a40f94b4b821.png">

Reply:
I believe the correct meaning of All/Active/Visible is as follows:
All: Any view registered to the view container
Active: Must be in All and When clause is true
Visible: Must in Active and be checked visible in the triple dot

Reply:
Wish I understood how I got into that state before though.

Reply:
same, perhaps it was related to multiwindow?